combat/CombatSystem.cpp

```diff
@@ -889,10 +889,18 @@ namespace {
             // do actual attacks
             Attack(attacker, *weapon_it, target, combat_state.combat_info, bout, round);
 
-            // mark attackers as valid targets for attacked object's owners, so
-            // attacker they can be counter-attacked in subsequent rounds if it
+            // mark attacker as valid target for attacked object's owner, so that regardless
+            // of visibility the attacker can be counter-attacked in subsequent rounds if it
             // was not already attackable
             combat_state.empire_infos[target->Owner()].target_ids.insert(attacker->ID());
+            // Also ensure that attacker (and their fleet if attacker was a ship) are
+            // revealed with at least BASIC_VISIBILITY to the taget empire
+            combat_state.combat_info.empire_object_visibility[target->Owner()][attacker->ID()] = 
+                    std::max(GetUniverse().GetObjectVisibilityByEmpire(attacker->ID(), target->Owner()), VIS_BASIC_VISIBILITY);
```

Comment:
As far as I could tell, it appeared these combat_info visibility maps had never been populated at all before.  Even if I overlooked that somewhere, this was the safest approach.  I had been trying to decide about putting in a warning comment about that, but decided to wait and do some more searching, but I suppose I should have added some kind of advisory/todo comment somewhere.  Mitten's version hadn't needed to do this checking because he was setting visibility to Partial, and so didn't have to worry that a non-owner might already have had a higher visibility.   If we change the later dissemination stage to check visibility and go with the greater, rather than just overwriting, then we wouldn't need to query it here.